File: Solutions/1. Two Sum/Solution.cs
using System.Collections.Generic;

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // Dictionary to store number and its index
        Dictionary<int, int> numToIndex = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];
            
            if (numToIndex.ContainsKey(complement)) {
                // Found the two numbers
                return new int[] { numToIndex[complement], i };
            }
            
            // Store the current number and its index
            numToIndex[nums[i]] = i;
        }
        

        return new int[0];
    }
}

File: Solutions/1. Two Sum/Solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut num_to_index = HashMap::new(); // HashMap to store number -> index
        
        for (i, &num) in nums.iter().enumerate() {
            let complement = target - num;
            
            if let Some(&index) = num_to_index.get(&complement) {
                // Found the two numbers
                return vec![index as i32, i as i32];
            }
            
            num_to_index.insert(num, i);
        }
        
        vec![]
    }
}
